src/modules/dashboard/components/Dialogs/AddExistingSafeDialog/AddExistingSafeDialogForm.tsx

```diff
@@ -120,11 +130,34 @@ const AddExistingSafeDialogForm = ({
   useEffect(() => {
     // If there's a valid address in the address input
     if (addressTouched && !addressError) {
-      // Get safe data
-      setIsLoadingSafe(true);
-      getSafeData(`${baseURL}api/v1/safes/${address}/`);
+      /*
+       * Check if safe already exists in colony
+       */
+      if (
+        colonySafes.filter((safe) => {
+          return (
+            safe?.chainId === selectedChain.value &&
+            safe?.contractAddress === address
+          );
+        }).length > 0
```

Comment:
I think something like this would be more readable:
```
const isSafeAlreadyAdded = !colonySafes.find(safe => 
  safe?.chainId === selectedChain.value && safe?.contractAddress === address
);

if (isSafeAlreadyAdded) { ...
```

What do you think?

Reply:
Yes like it (but without the negation as that would be if safe is not already added).